pkg/app/web/src/components/application-form/index.tsx

```diff
@@ -425,3 +437,211 @@ export const ApplicationForm: FC<ApplicationFormProps> = memo(
+    };
+
+    return (
+      <div className={classes.inputGroup}>
+        <FormControl className={classes.formItem} variant="outlined">
+          <InputLabel id="filter-piped">Piped</InputLabel>
+          <Select
+            labelId="filter-piped"
+            id="filter-piped"
+            label="PipedId"
+            className={classes.select}
+            onChange={(e) => {
+              setSelectedPipedId(e.target.value as string);
+              handleUpdateFilterValue({
+                pipedId: (e.target.value as string)
+              });
+            }}
+          >
+            {pipeds.map((e) => (
+              <MenuItem value={e.id} key={`piped-${e.id}`}>
+                {e.name}
+              </MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+        <div className={classes.inputGroupSpace} />
+        <FormControl className={classes.formItem} variant="outlined">
+          <InputLabel id="filter-cloud-provider">Cloud Provider</InputLabel>
+          <Select
+            labelId="filter-cloud-provider"
+            id="filter-cloud-provider"
+            label="CloudProvider"
+            className={classes.select}
+            disabled={selectedPipedId === ""}
+            onChange={(e) => {
+              const values = (e.target.value as ReadonlyArray<string>);
+              handleUpdateFilterValue({
+                cloudProvider: values[0],
+                kind: values[1],
+                pipedId: selectedPipedId,
+              });
+            }}
+          >
+            {cloudProviders.map((e) => (
+              <MenuItem value={([e.name, e.type] as ReadonlyArray<string>)} key={`cloud-provider-${e.name}`}>
+                {e.name}
+              </MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+      </div>
+    );
+  }
+);
+
+const UnregisteredApplicationList: FC<ApplicationFormProps> = memo(
+  function ApplicationForm({
+    onClose,
+    onAddFromGit,
+  }) {
+    const classes = useStyles();
+    // TODO: Call api to fetch all unregisteredApps
+    /*const apps = useAppSelector<ApplicationInfo.AsObject[]>((state) =>
+        fetchUnregisteredApplications()
+    );*/
+    const apps: ApplicationInfo.AsObject[] = [];
+
+    const [selectedPipedId, setSelectedPipedId] = useState("");
+    const [selectedKind, setSelectedKind] = useState("");
+
+    const handleFilterChange = useCallback(
+      (options: UnregisteredApplicationsFilterOptions) => {
+        setSelectedPipedId(options.pipedId);
+        setSelectedKind(options.kind);
+        // TODO: Set cloud provider as well
+      },
+      [],
+    );
+    return (
+      <Box width="100%">
+        <Typography className={classes.title} variant="h6">
+          Select the application to register
+        </Typography>
+        <Divider />
+        <Box width="100%" m={0} px={2}>
+          <UnregisteredApplicationFilter
+            onChange={handleFilterChange}
+          />
+          <Accordion disabled>
```

Comment:
nits, add a little space between this and the filters group above make it looks better 😉

Reply:
I feel you. I'm now stuck with that 😵‍💫

Reply:
According the API https://mui.com/api/accordion/#props
we can pass `<Accordion className={classes.xxx}` here and define the padding there to make the CSS affect only this element 👀

Reply:
Thank you. I applied it!

Reply:
It looks like:
![image](https://user-images.githubusercontent.com/19730728/144984535-b3989abb-b6a1-43a1-accc-5b80854dfa8a.png)

Reply:
Nice thank you 🙌